public/app/plugins/panel/canvas/migrations.ts

```diff
@@ -25,9 +25,15 @@ export const canvasMigrationHandler = (panel: PanelModel): Partial<Options> => {
       for (const element of root.elements) {
         if (element.type === 'ellipse') {
           // Take existing ellipse specific background and border config and apply it to the element's general background and border config
-          element.background = element.config.backgroundColor;
-          element.border.color = element.config.borderColor;
-          element.border.width = element.config.width;
+          if (element.config.backgroundColor) {
+            element.background = element.config.backgroundColor;
+          }
+          if (element.config.borderColor) {
+            element.border.color = element.config.borderColor;
+          }
+          if (element.config.width) {
+            element.border.width = element.config.width;
+          }
 
           // Remove the ellipse specific background and border config
           delete element.config.backgroundColor;
```

Comment:
Should we also throw these inside the if statements?

Reply:
yeah, I was thinking about it, I'll do

Reply:
Might as well. 😄